Feature/tx query graph

no-reply opened the pull request:
This would close #269 and #278. The failing specs in CI call out a pre-existing issue with RDF::Graph's Queryable support that needs to be resolved before merge.

This changes the Transaction semantics to behave like Graph. Queryable is somewhat opinionated that inserting triples to a queryable should result in those triples being accessible through query. In light of this, the answer to #278 seems to be that graph scope must be symmetrical for read and write operations. That interpretation is implemented in the PR.

The remaining blocker is: Queryable's shared examples require a query pattern's graph name to be retained when querying, but Graph overwrites it.

My inclination is to change the specs (skip when `#graph_name` is defined), keeping the existing behavior. Thoughts, @gkellogg?

gkellogg commented:
That seems reasonable to me. Please be sure SPARQL specs pass with this change, though.

no-reply commented:
Fixed up the specs, squashed out the `Gemfile` changes, and force pushed.

https://github.com/ruby-rdf/rdf-spec/pull/59 is the corresponding spec update.

A local run of `SPARQL` with these branches passes.
